src/TitleDetails/AddPieceModal/AddPieceModal.js

```diff
@@ -0,0 +1,199 @@
+import React, { useCallback } from 'react';
+import { FormattedMessage } from 'react-intl';
+import PropTypes from 'prop-types';
+import { Field } from 'react-final-form';
+import {
+  get,
+  includes,
+} from 'lodash';
+
+import {
+  Button,
+  Checkbox,
+  Col,
+  Modal,
+  ModalFooter,
+  Row,
+  TextArea,
+  TextField,
+} from '@folio/stripes/components';
+import stripesFinalForm from '@folio/stripes/final-form';
+import {
+  FieldDatepickerFinal,
+  FieldSelectFinal,
+  validateRequired,
+} from '@folio/stripes-acq-components';
+
+import {
+  INVENTORY_RECORDS_TYPE,
+} from '../constants';
+import AddItemPlugin from '../AddItemPlugin';
+
+const AddPieceModal = ({
+  close,
+  createInventoryValues,
+  form,
+  handleSubmit,
+  instanceId,
+  locations,
+  onCheckIn,
+  pieceFormatOptions,
+}) => {
+  const addItem = form.mutators.setItemValue;
+  const formValues = get(form.getState(), 'values', {});
+  const { format, id, itemId, locationId } = formValues;
+  const isLocationRequired = includes(createInventoryValues[format], INVENTORY_RECORDS_TYPE.instanceAndHolding);
+  const isAddItemRequired = includes(createInventoryValues[format], INVENTORY_RECORDS_TYPE.all);
+  let isAddItemButtonDisabled = true;
+
+  if (!itemId && locationId && isAddItemRequired) {
+    isAddItemButtonDisabled = false;
+  }
+
+  const disabledButtonProps = isAddItemButtonDisabled ? { disabled: isAddItemButtonDisabled } : {};
+  const labelId = id ? 'ui-receiving.piece.addPieceModal.editTitle' : 'ui-receiving.piece.addPieceModal.title';
+
+  const receive = useCallback(
+    () => onCheckIn(formValues),
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
what is the reason of this comment here? shouldn't be added `onCheckIn` as a deb as well?